### PR TITLE
Added support for remote OSX 

### PR DESCRIPTION
**Pull request by ronenzaslaver**
Verified bbtest tests pass on Win, Linux and OSX

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 from tests.test_os_agnostic import BaseToDoTest
 
 os_2_class = {'local': LocalHost,
               'win': WindowsHost,
               'windows': WindowsHost,
               'win32': WindowsHost,
-              'linux': LinuxHost}
+              'linux': LinuxHost,
+              'mac': OSXHost,
+              'osx': OSXHost}
```
Comment (daonb) [inline tests/conftest.py]: "There should be one-- and preferably only one --obvious way to do it."

Please join 'mac' & 'osx' into one entry and call it `posix`  https://stackoverflow.com/a/1879/66595

File: tests/test_lab.py [hunk 2]
```diff
@@ -16,6 +17,26 @@ def test_no_lab(self):
         pass
 
 
+class TestNoClass(BBTestCase):
+
```
Comment (daonb) [inline tests/test_lab.py]: This test code is doing something special and I'm not sure I follow. Please add a comment explaining what it tests.

File: tests/conftest.py (hunk 1 above)
Comment (ronenzaslaver) [reply tests/conftest.py]: need to further discuss it. maybe requires a more extensive fix.

File: tests/test_lab.py (hunk 2 above)
Comment (daonb) [reply tests/test_lab.py]: @shmir I'm merging the PR, but please add a comment

**Pull request merged**